contracts/schemes/GlobalConstraintRegistrar.sol

```diff
@@ -29,39 +29,48 @@ contract GlobalConstraintRegistrar is VotingMachineCallbacks, ProposalExecuteInt
 
     event ProposalExecuted(address indexed _avatar, bytes32 indexed _proposalId, int256 _param);
     event ProposalDeleted(address indexed _avatar, bytes32 indexed _proposalId);
-    event D(bool x);
 
     // The struct that holds the information of a global constraint proposed to be added or removed.
     struct GCProposal {
         address gc; // The address of the global constraint contract.
         bool addGC; // true: add a GC, false: remove a GC.
-        bytes32 voteToRemoveParams; // Voting parameters for removing this GC.
     }
 
     // GCProposal by avatar and proposalId
     mapping(bytes32=>GCProposal) public organizationProposals;
 
-    // voteToRemoveParams hash by avatar and proposal.gc
-    mapping(address=>bytes32) public voteToRemoveParams;
-
     /**
      * @dev initialize
      * @param _avatar the avatar this scheme referring to.
-     * @param _votingMachine the voting machines address to
-     * @param _votingParams genesisProtocol parameters - valid only if _voteParamsHash is zero
-     * @param _voteOnBehalf genesisProtocol parameter - valid only if _voteParamsHash is zero
-     * @param _voteParamsHash voting machine parameters.
+     * @param _votingParams genesisProtocol parameters
+     * @param _voteOnBehalf  parameter
+     * @param _daoFactory  DAOFactory instance to instance a votingMachine.
+     * @param _stakingToken (for GenesisProtocol)
+     * @param _packageVersion packageVersion to instance the votingMachine from.
+     * @param _votingMachineName the votingMachine contract name.
      */
     function initialize(
         Avatar _avatar,
-        IntVoteInterface _votingMachine,
         uint256[11] calldata _votingParams,
         address _voteOnBehalf,
-        bytes32 _voteParamsHash
+        DAOFactory _daoFactory,
+        address _stakingToken,
+        uint64[3] calldata _packageVersion,
+        string calldata _votingMachineName
     )
     external
     {
-        super._initializeGovernance(_avatar, _votingMachine, _voteParamsHash, _votingParams, _voteOnBehalf);
+        super._initializeGovernance(
+            _avatar,
+            _votingParams,
+            _voteOnBehalf,
+            _daoFactory,
+            _stakingToken,
+            address(this),
+            address(this),
+            address(this),
```

Comment:
If in all our schemes it's all the scheme address 3 times why not just make a `scheme` parameter for all the 3 instead?

Anyway this seems wrong actually, shouldn't this be the Avatar address, then twice the scheme address, instead of 3 times the scheme?

This comment is for all schemes...